rust/common/error.rs

```diff
@@ -140,54 +100,151 @@ macro_rules! max {
     ($x:literal, $($xs:literal),*) => (max($x, max!($($xs),*)));
 }
 
+macro_rules! format_message {
+    ($self:ident, $error_name:ident, $body:literal, ) => {
+        format!($body)
```

Comment:
`format!()` enables the argument checking. This will not compile if `$body` contains any kind of formatting that expects arguments (i.e. `{}`)